samples/net/mqtt_azure/README.rst

```diff
@@ -0,0 +1,87 @@
+.. _mqtt-azure-sample:
+
+MQTT Azure Sample
+#################
+
+Overview
+********
+
+This sample application demonstrates how an MQTT client
+can publish messages to an Azure Cloud IoT hub based on MQTT protocol.
+
+- Acquire a DHCPv4 lease
+- Establish a TLS connection with Azure Cloud IoT hub
+- Publish data to the Azure cloud
+- SOCKS5 supported
+
+The source code of this sample application can be found at:
+:zephyr_file:`samples/net/mqtt-azure`.
+
+Requirements
+************
+
+- Azure Cloud account
+- Azure IOT Cloud credentials and required information
+- Freedom Board (FRDM-K64F)
+- Network connectivity
+
+Building and Running
+********************
+This application has been built and tested on the NXP FRDMK64F.
+Certs are required to authenticate to the Azure Cloud IoT hub.
+Current certs in `samples/net/mqtt_azure/src/test_certs.h` are
+copied from `<https://github.com/Azure/azure-iot-sdk-c/blob/master/certs/certs.c>`_
+
+Configure the following Kconfig options based on your Azure Cloud IoT Hub:
+
+- SAMPLE_CLOUD_AZURE_USERNAME - Username field use::
```

Comment:
This doesn't render in HTML quite as you'd expect.  (Check out the doc build artifacts at https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/20461/docs/samples/net/mqtt_azure/README.html#building-and-running). Rewrite this first bullet item as:
```
- SAMPLE_CLOUD_AZURE_USERNAME - Username field use::

     {iothubhostname}/{device_id}/?api-version=2018-06-30

  where ``{iothubhostname}`` is the full CName of the IoT hub.
```